packages/env/lib/commands/start.js

```diff
@@ -203,17 +203,16 @@ module.exports = async function start( {
 			} ),
 		] );
 
-		// Execute any configured command that should run after the environment has finished being set up.
-		if ( scripts ) {
-			executeAfterSetup( config, spinner );
-		}
-
 		// Set the cache key once everything has been configured.
 		await setCache( CONFIG_CACHE_KEY, configHash, {
 			workDirectoryPath,
 		} );
 	}
 
+	if ( scripts ) {
+		await executeLifecycleScript( 'afterStart', config, spinner );
```

Comment:
I'm torn on this one. Ultimately, I'd personally prefer to only run it during configuration, because we'd loose the performance benefit we get from having a "no-work" startup mode. However, that's a lot harder to communicate to users and it's more confusing. :/

Reply:
I was too, but, I ended up in the same place. Even `afterSetup` felt really ambiguous in the first place.